app/main.py

```diff
@@ -0,0 +1,52 @@
+from functools import lru_cache
+from logging import basicConfig, INFO, DEBUG, getLogger
+
+from fastapi import FastAPI
+from uvicorn import run
+
+from app.routers.redirect import redirect_router
+from app.routers.links import links_router
+from app.services.geo_ip.wrapper import GeoIP2Wrapper
+from app.services.keycloak.wrapper import KeycloakWrapper
+from app.services.mongodb.wrapper import MongoDBWrapper
+from app.settings import get_settings
+
+settings = get_settings()
+basicConfig(level=DEBUG if settings.debug else INFO)
+logger = getLogger(__name__)
+
+
+def add_event_handlers(app: FastAPI) -> None:
+    mongodb_wrapper = MongoDBWrapper()
+    keycloak_wrapper = KeycloakWrapper()
+    geoip2_wrapper = GeoIP2Wrapper()
+    app.add_event_handler("startup", mongodb_wrapper.startup_event_handler)
+    app.add_event_handler("shutdown", mongodb_wrapper.shutdown_event_handler)
+    app.add_event_handler("startup", keycloak_wrapper.startup_event_handler)
+    app.add_event_handler("shutdown", keycloak_wrapper.shutdown_event_handler)
+    app.add_event_handler("startup", geoip2_wrapper.startup_event_handler)
+    app.add_event_handler("shutdown", geoip2_wrapper.shutdown_event_handler)
+
+
+def add_routers(app: FastAPI) -> None:
+    app.include_router(router=links_router, prefix="/shorten")
+    app.include_router(router=redirect_router)
+
+
+def add_middlewares(app: FastAPI) -> None:  # pylint: disable=unused-argument
+    ...
+
+
+@lru_cache
+def create_app() -> FastAPI:
+    app = FastAPI(title=settings.title, version=settings.version)
```

Comment:
You could also pass `debug` here:

```suggestion
    app = FastAPI(title=settings.title, version=settings.version, debug=settings.debug)
```

Dunno if it makes sense for your use case.

Reply:
+